cartography/intel/rapid7/util.py

```diff
@@ -0,0 +1,246 @@
+"""
+cartography/intel/rapid7/util
+"""
+import json
+import logging
+import re
+from typing import Any
+from typing import List
+from typing import Tuple
+
+import pandas
+import requests
+
+logger = logging.getLogger(__name__)
+
+
+def extract_sub_from_resourceid(resourceid: str) -> str:
+    """
+    Get Azure subscription from Azure resource id
+    """
+    if isinstance(resourceid, str):
+        match = re.search(
+            "/subscriptions/([0-9a-f-].*?)/resourceGroups/",
+            resourceid,
+            flags=re.IGNORECASE,
+        )
+        if match:
+            sub = match.group(1)
+            return sub
+    return ""
+
+
+def extract_rg_from_resourceid(resourceid: str) -> str:
+    """
+    Get Azure subscription from Azure resource id
+    """
+    if isinstance(resourceid, str):
+        match = re.search(
+            "/resourceGroups/(.*?)/providers/",
+            resourceid,
+            flags=re.IGNORECASE,
+        )
+        if match:
+            resource_group = match.group(1)
+            return resource_group
+    return ""
+
+
+def extract_rapid7_configurations_subscriptionid(configurations: Any[str, list]) -> str:
+    """
+    Extract from json, specific keyvalue pair: azure subscription id
+
+    Input: dataframe row configurations column
+    """
+
+    if configurations is None or not isinstance(configurations, list):
+        return ""
+
+    subscription_id = ""
+    for line in configurations:
+        if line["name"] == "azure":
+            logger.warning("line: %s", line)
+            if isinstance(line["value"], str):
+                azure = json.loads(line["value"])
+            else:
+                azure = line["value"]
+            subscription_id = extract_sub_from_resourceid(azure["resourceId"])
+            logger.warning(
+                "extract_rapid7_configurations_subscriptionid0: %s (%s)",
+                subscription_id,
+                azure["resourceId"],
+            )
+
+    logger.warning("extract_rapid7_configurations_subscriptionid: %s", subscription_id)
+    return subscription_id
+
+
+def extract_rapid7_configurations_resourcegroup(configurations: Any[str, list]) -> str:
+    """
+    Extract from json, specific keyvalue pair: azure resource group
+
+    Input: dataframe row configurations column
+    """
+
+    if configurations is None or not isinstance(configurations, list):
+        return ""
+
+    resource_group = ""
+    for line in configurations:
+        if line["name"] == "azure":
+            logger.debug("line: %s", line)
+            if isinstance(line["value"], str):
+                azure = json.loads(line["value"])
+            else:
+                azure = line["value"]
+            resource_group = extract_rg_from_resourceid(azure["resourceId"])
+
+    logger.warning("extract_rapid7_configurations_resourcegroup: %s", resource_group)
```

Comment:
Why is this a warning? If resouregroup would produce too many messages, then perhaps it's more appropriate to have this as a debug message.